src/Symfony/Component/HttpClient/UriTemplateHttpClient.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpClient;
+
+use Symfony\Component\HttpClient\UriTemplate\CallbackUriTemplateExpander;
+use Symfony\Component\HttpClient\UriTemplate\UriTemplateExpanderInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+use Symfony\Contracts\HttpClient\ResponseStreamInterface;
+
+class UriTemplateHttpClient implements HttpClientInterface
+{
+    private HttpClientInterface $client;
+    private UriTemplateExpanderInterface $expander;
+
+    private array $variables = [];
+
+    public function __construct(
+        HttpClientInterface $client = null,
+        UriTemplateExpanderInterface $expander = null,
+    ) {
+        $this->client = $client ?? HttpClient::create();
+        $this->expander = $expander ?? CallbackUriTemplateExpander::createFromPopularVendors();
+    }
+
+    public function request(string $method, string $url, array $options = []): ResponseInterface
+    {
+        if ($this->variables && null !== ($options['base_uri'] ?? null)) {
+            $options['base_uri'] = $this->expander->expand($options['base_uri'], $this->variables);
+        }
+
+        return $this->client->request(
+            $method,
+            $this->variables ? $this->expander->expand($url, $this->variables) : $url,
```

Comment:
the expander should also be called when no variables are set, so that default values can be resolved in templates (same everywhere it's called)

Reply:
Which default values?

Reply:
hum nvm, those don't exist :grimacing: